Switch Dashboard to Hub's project data

gboone opened the pull request:
One place to update project data _forever_ (or at least until we change it again). #iterating

Commit:
Switch to using 18F/data-public submodule

Commit:
Switch to using 18F/data-public submodule

afeld committed:
add contributor notes about running site and making data changes

afeld committed:
add contributor notes about running site and making data changes

afeld committed:
add gemfile

afeld committed:
add gemfile

afeld committed:
build all branches

afeld committed:
build all branches

afeld committed:
initialize submodules on travis

afeld committed:
initialize submodules on travis

afeld committed:
add a bootstrap script

afeld committed:
add a bootstrap script

afeld committed:
travis automatically handles submodules

afeld committed:
travis automatically handles submodules

afeld committed:
use https url for submodule

afeld committed:
use https url for submodule

afeld committed:
add an update script

afeld committed:
add an update script

afeld committed:
update data

afeld committed:
specify branch to use for data submodule

afeld committed:
specify branch to use for data submodule

afeld committed:
update data

afeld committed:
point to data-public documentation

afeld committed:
point to data-public documentation

Commit:
Adds submodule updating to the deploy fabfile

Commit:
Adds submodule updating to the deploy fabfile

Commit:
Merge branch 'data-submodule' of github.com:18F/dashboard into data-submodule

Commit:
Removes `data-public` module

Commit:
Should update gitmodule to use Hub's public data

Commit:
Adds a ci specific script that Travis will use.

- Requires a projects.yml file committed to the repo

Commit:
Ignore submodules

mbland commented on script/ci:
```diff
@@ -0,0 +1,9 @@
+bundle
+test -e _data/projects.yml
+
+jekyll build
```
Might wanna `bundle exec jekyll build` here.

mbland commented on CONTRIBUTING.md:
````diff
@@ -1,6 +1,18 @@
+```bash
+./script/bootstrap
+bundle exec jekyll serve -w
+open http://localhost:3000/dashboard/
````
Should be `localhost:4000`, no?

Commit:
Replace the shell scripts with a go script

Cuz go scripts are cool. Now just run ./go [thing you want to do] (as long
as the thing you want to do is either `init`, `update_gems`, `submodules`,
`serve`, `build` or `update_data`.

Commit:
Udates readme with details about use the go script

Commit:
Switches CONTRIBUTING docs to use go scripts

Commit:
Tell travis to use the go scripts.

Commit:
Clearer instructions on when/who should submodule

Commit:
Remove the no-longer-existed submodule cmd

Commit:
fixup! Remove the no-longer-existed submodule cmd

Commit:
Adds a ci_build to the go script

Commit:
No more data-public, for the last time

Commit:
fixup! No more data-public, for the last time
